### PR TITLE
Add support for OpenTelemetry

### PR DESCRIPTION
**Pull request by dhoard**
@fstab please review.

### CODE REVIEW

**Comment by zeitlinger** [comment]
@dhoard that's a big pr - can you walk me through it on the next meeting or set up a call?

**Comment by dhoard** [comment]
> @dhoard that's a big pr - can you walk me through it on the next meeting or set up a call?

Definitely.

File: jmx_prometheus_common/src/main/java/io/prometheus/jmx/common/opentelemetry/OpenTelemetryExporterFactory.java [hunk 1]
```diff
@@ -0,0 +1,192 @@
...
+                                        new ValidateIntegerInRange(
+                                                1,
+                                                Integer.MAX_VALUE,
+                                                ConfigurationException.supplier(
+                                                        "Invalid configuration for"
+                                                                + " /openTelemetry/interval must be"
+                                                                + " an integer greater than 0")))
+                                .orElse(NO_INTERVAL);
+
+                Map<String, String> headers =
```
Comment (zeitlinger) [inline jmx_prometheus_common/src/main/java/io/prometheus/jmx/common/opentelemetry/OpenTelemetryExporterFactory.java:126]: would be good to support resource attributes the same way

File: jmx_prometheus_javaagent/src/test/java/io/prometheus/jmx/ArgumentsTest.java [hunk 2]
```diff
@@ -0,0 +1,333 @@
...
+                INVALID_CONFIGURATION,
+                "::/opt/prometheus/config.yaml",
+                Arguments.Mode.OPEN_TELEMETRY,
+                null,
+                null,
+                "/opt/prometheus/config.yaml")*/
+    };
+
+    @Test
+    public void testsArguments() {
```
Comment (zeitlinger) [inline jmx_prometheus_javaagent/src/test/java/io/prometheus/jmx/ArgumentsTest.java:281]: parmeterized test is easier to troubleshoot

see https://junit.org/junit5/docs/current/user-guide/#writing-tests-parameterized-tests-sources-MethodSource
and https://github.com/grafana/grafana-opentelemetry-java/blob/50411a43bf63896d4f68f1634d5a047e365399d2/custom/src/test/java/com/grafana/extensions/resources/ResourceCustomizerTest.java for an example

Comment (dhoard) [reply jmx_prometheus_javaagent/src/test/java/io/prometheus/jmx/ArgumentsTest.java]: Fixed.

**Pull request merged**